### ISSUE TITLE
add otel-cli status subcommand

### ISSUE THREAD

**Issue by tobert** [equinix-labs/otel-cli#63]
Write a subcommand that loads traceparent, configs, all that, and prints its state to stdout.

I prototyped this for otel-init-go in https://github.com/equinix-labs/otel-init-go/blob/main/cmd/test-otel-init-go/main.go which it uses to dump state. This functionality has similar value here, for testing, and for users who just want to know what otel-cli thinks its configuration is.

Side quest: move the config globals into a config struct, which would make the code for the command straightforward and easier to maintain since it will automatically pick up new fields in the config struct.

Output e.g. (taken from the otel-init-go stub, will likely reflect otel-cli's guts more)

```json
{
    "config": {
        "endpoint": "",
        "insecure": "false",
        "service_name": "otel-cli"
    },
    "env": {
        "PATH": "/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin"
    },
    "otel": {
        "is_sampled": "false",
        "span_id": "0000000000000000",
        "trace_flags": "00",
        "trace_id": "00000000000000000000000000000000"
    }
}
```

**Comment by tobert** [comment]
I'm torn about this. Super useful for testing, but a little dangerous on the opsec side:  as a user I might want to `otel-cli status` in my CI pipeline so it's captured in the output for debugging. For authenticated users we could end up helping people accidentally print tokens into their logs and that's not cool.

Maybe it should be clearly named e.g. `otel-cli unsafe-dump-status`? Still useful for admins but clearly not safe for CI?

**Comment by tobert** [comment]
Is there a way I can get ahold of the gRPC server status? That would be so rad for debugging OTLP.

**Comment by tobert** [comment]
Also maybe just add `--dump-config` to `otel-cli span`? That would fire up a connection to the server and try to send a span that could then be printed for making sure everything actually works. And perhaps a separate `otel-cli dump-config` that just prints what it knows and does not try to connect. 🤔

**Comment by ahayworth** [comment]
> For authenticated users we could end up helping people accidentally print tokens into their logs and that's not cool.

You could always default to obfuscating the tokens (and the like) in the debug output, and then use a flag to un-obfuscate it on request. `otel-cli status --reveal-to-me-your-secrets` or something?

**Comment by tobert** [comment]
Shipped in v0.0.15.
